chainerx_cc/chainerx/routines/indexing.cc

```diff
@@ -292,4 +292,16 @@ Array Where(const Array& condition, Scalar x, Scalar y) {
     return out;
 }
 
+Array Nonzero(const Array& a) {
+    Array condition = a != 0.0;
+    // Initialize out with appropriate shape
```

Comment:
Uncertain about, if it is the best way. But you can start with this.
You should find the number of non-zero elements something like this 
```CPP
int64_t nonzerocount = (a == 0).Sum()
// allocate `n` output arrays of `nonzerocount` elements 
// where `n` is dimension of `a`.
```

Reply:
@aksub99 Ooops. You are right I meant, `a!=0`.

n-array allocations because you return indices of non-zero elements.
i.e. for 2-dim array, one array for `x-indices` and other for `y-indices`. 
Reference: https://docs.scipy.org/doc/numpy/reference/generated/numpy.nonzero.html

Reply:
Oh yes! Thanks for your confirmation @kshitij12345 .

Reply:
Some dig up into the `numpy` code,

Numpy uses a similar approach, of first counting the non-zero elements and allocating the array.
https://github.com/numpy/numpy/blob/0dcd3001eb80a33a60127467fff84747d98c39c8/numpy/core/src/multiarray/item_selection.c#L2249-L2265

However they allocate single 2-d array of shape `(ndim, nonzerocount)` and  split the 2-d array i.e. at the end return `n` views to it each of size `non_zerocount`
where `n` is the dimension of input array.
https://github.com/numpy/numpy/blob/0dcd3001eb80a33a60127467fff84747d98c39c8/numpy/core/src/multiarray/item_selection.c#L2414-L2431

Reply:
@asi1024 Would it be better if `chainerx::nonzero` returns a tuple similar to the numpy behavior?
If so, should `out` be initialized as an empty tuple?

Reply:
The size of `std::tuple` must be determined compilation time, but